### PR TITLE
Instrument_list query mod to left join on test battery

### PR DESCRIPTION
**Pull request by ridz1208**

### CODE REVIEW

**Comment by tedstrauss** [comment]
I have tested this on a CCNA battery. The test battery loads identically.

**Comment by driusan** [comment]
@stellarxo can you check if this fixes your problem where tests weren't showing up if they aren't in the test_battery?

File: php/libraries/NDB_BVL_Battery.class.inc [hunk 1]
```diff
@@ -312,30 +312,18 @@ class NDB_BVL_Battery
...
             AND LEFT(f.CommentID, 3) != 'DDE'";
 
-        if (!is_null($stage)) {
-            $query .= " AND t.Test_name=b.Test_name";
-
-            if (!is_null($SubprojectID)) {
-                $query .= " AND (
-                    b.SubprojectID IS NULL
-                    OR b.SubprojectID=:SubprojID
-                )";
```
Comment (driusan) [inline php/libraries/NDB_BVL_Battery.class.inc]: I don't think this if statement should be removed. It completely changes the behaviour and will break scripts like assign missing instruments.php

**Comment by stellarxo** [comment]
500 error? Cannot access timepoint list

**Comment by stellarxo** [comment]
[Fri Jul 22 09:07:27.123230 2016] [:error] [pid 9057] [client 132.216.232.122:58440] PHP Fatal error:  Call to undefined method NDB_Menu_Filter_instrument_list::_assertBatterySelected() in /var/www/loris/modules/instrument_list/php/NDB_Menu_Filter_instrument_list.class.inc on line 150

**Comment by stellarxo** [comment]
Tested and it works!

File: php/libraries/NDB_BVL_Battery.class.inc [hunk 2]
```diff
@@ -312,30 +312,24 @@ class NDB_BVL_Battery
...
-        $query  .= " FROM flag AS f, test_names as t, test_subgroups as ts ";
+        $query  .= " FROM flag f
+            	JOIN test_names t ON (f.Test_name=t.Test_name)
+            	JOIN test_subgroups ts ON (ts.ID = t.Sub_group) ";
         $qparams = array('SID' => $this->sessionID);
 
         if (!is_null($stage)) {
-            $query .= ", test_battery AS b";
+            $query .= "
+            	LEFT JOIN test_battery b ON (t.Test_name=b.Test_name) ";
```
Comment (stellarxo) [inline php/libraries/NDB_BVL_Battery.class.inc]: Should this still be edited if you already put a new function above?
Other things might depend on this specific behaviour

Comment (ridz1208) [inline php/libraries/NDB_BVL_Battery.class.inc]: This is a better way of doing the query.

Comment (stellarxo) [inline php/libraries/NDB_BVL_Battery.class.inc]: Dave agrees that the LEFT JOIN is better :)

**Pull request merged**